### ISSUE TITLE
BT-33 legalinfo

### ISSUE THREAD

**Issue by TeXhackse** [TeXhackse/LaTeX-ZUGFeRD#33]
Regarding the 'additional legal information' (BT-33) referenced in chapter 13.3 of the 'Spezifikation Standard XRechnung' cited above, a Zugferd key 'seller/adlegalinfo' could be introduced. I thought that something like this could work ...

```
diff --git a/zugferd.dtx b/zugferd.dtx
index 15bdb33..a8f427e 100644
--- a/zugferd.dtx
+++ b/zugferd.dtx
@@ -1669,6 +1669,9 @@
        \int_gincr:N \g_@@_indent_int%
        \_@@_write_xml:e {%
                <ram:Name>\prop_item:Nn \g_@@_seller_AddressData_prop {name}</ram:Name>
+               \tl_if_blank:nF {\prop_item:Nn \g_@@_seller_AddressData_prop {adlegalinfo}}{%
+               <ram:Description>\prop_item:Nn \g_@@_seller_AddressData_prop {adlegalinfo}</ram:Description>
+               }%
                \_@@_DefinedTradeContact:N \g_@@_seller_AddressData_prop%
                \_@@_PostalTradeAddress:N \g_@@_seller_AddressData_prop%
        }%
```
... and found that the code includes the value of seller/adlegalinfo as expected, but if the key is empty it still includes <ram:Description></ram:Description> in the xml output.

_Originally posted by @knupero in https://github.com/TeXhackse/LaTeX-ZUGFeRD/issues/31#issuecomment-2517629568_

**Comment by TeXhackse** [comment]
@knupero: This does not work as you are testing for a blank token list, which is obviously not blank. 

```
\tl_if_blank:nF {\prop_item:Nn \g_@@_seller_AddressData_prop {adlegalinfo}}{…}
```
is always false, as the first argument contains something which is not a blank.  one would check if the property contains the actual item instead.

**Comment by knupero** [comment]
After well over 30 years of experience with TeX, this is really the very first time that I have dealt with latex3 code in any way. Somehow this is a completely different world, in which Knuth's TeXbook, which is lying here on my desk, hardly helps any more, and in which old practiced reflexes to solve problems with \def, \edef, \expandafter etc. have become obsolete and counterproductive ;-) I'll try to fix that.

**Comment by TeXhackse** [comment]
No… that's not what I wanted. Actually I tried to explain… But this is not what these issue and this project is here for… Either you know what you code snippet does … As I actualy want to use my time on working on that project instead of explaining you the structure. That's what I did e.g. in https://tug.org/TUGboat/tb44-1/tb136peischl-expl3.html (german variant at http://archiv.dante.de/DTK/PDF/komoedie_2022_4.pdf#page=24 and french translation would also be available at https://publications.gutenberg-asso.fr/lettre/article/view/153 )


Back to topic. And no the issue is not the code, the thing which takes me time is the documentation and thinking about matching the interface … 

Anyway as I already started … : Either you expand the second argument exhaustive before you run the test: 
```
\tl_if_blank:eF {\prop_item:Nn \g_@@_seller_AddressData_prop {adlegalinfo}}{…}
```
Which you could… in some way describe as a more powerful version of `\expandafter`

But what I meant… there is an interface which is actually better, as we now use a tokenlist test to test a property list … One could just use

```
\prop_if_in:NnT \g_@@_seller_AddressData_prop {adlegalinfo} {…}
```
instead.

And that's a great example why expl3 is worth it. You don't have to construct things by conbining primitives.
